lib/mayaUsd/ufe/UsdUndoDuplicateCommand.cpp

```diff
@@ -92,12 +93,37 @@ void UsdUndoDuplicateCommand::execute()
     // state.
     duplicateLoadRules(*stage, path, _usdDstPath);
 
-    bool retVal = PXR_NS::SdfCopySpec(_srcLayer, path, _dstLayer, _usdDstPath);
-    TF_VERIFY(
-        retVal,
-        "Failed to copy spec data at '%s' to '%s'",
-        prim.GetPath().GetText(),
-        _usdDstPath.GetText());
+    // Make sure all necessary parent exists in the target layer, at least as over,
+    // otherwise SdfCopySepc will fail.
+    SdfJustCreatePrimInLayer(_dstLayer, _usdDstPath.GetParentPath());
+
+    // Retrieve the layers where there are opinion and order them from weak
+    // to strong. We will copy the weakest opinions first, so that they will
+    // get over-written by the stronger opinions.
+    using namespace MayaUsdUtils;
+    std::vector<LayerAndPath> authLayerAndPaths = getAuthoredLayerAndPaths(prim);
+    std::reverse(authLayerAndPaths.begin(), authLayerAndPaths.end());
```

Comment:
I don't think this has much importance here, but it would be more performance-efficient to not do std::reverse here and instead do a reverse cycle in the 'for' loop below.